src/Symfony/Component/Validator/Constraints/EmailValidator.php

```diff
@@ -78,13 +131,15 @@ public function validate($value, Constraint $constraint)
 
                 return;
             }
-        } elseif (!preg_match('/^.+\@\S+\.\S+$/', $value)) {
-            $this->context->buildViolation($constraint->message)
-                ->setParameter('{{ value }}', $this->formatValue($value))
-                ->setCode(Email::INVALID_FORMAT_ERROR)
-                ->addViolation();
+        } elseif (array_key_exists($constraint->mode, self::$emailPatterns)) {
```

Comment:
`elseif (isset(self::$emailPatterns[$constraint->mode]) && !preg_match(self::$emailPatterns[$constraint->mode], $value)) { }` ?

We still lack some mode normalization on the constraint (only default mode is normalized). Opens a small trap.

Also instead of normalizing.. i think i prefer an exception in case of unknown modes passed. Which seems the regular way of doing things; https://github.com/symfony/symfony/blob/master/src/Symfony/Component/HttpFoundation/Cookie.php#L131